lib/WiFiConnection/WifiConnection.cpp

```diff
@@ -0,0 +1,88 @@
+/*
+  WifiConnection - Helper for connecting to a wifi network
+  Project home: https://github.com/puuu/MQTT433gateway/
+
+  The MIT License (MIT)
+
+  Copyright (c) 2017 Jan Losinski
+
+  Permission is hereby granted, free of charge, to any person
+  obtaining a copy of this software and associated documentation files
+  (the "Software"), to deal in the Software without restriction,
+  including without limitation the rights to use, copy, modify, merge,
+  publish, distribute, sublicense, and/or sell copies of the Software,
+  and to permit persons to whom the Software is furnished to do so,
+  subject to the following conditions:
+
+  The above copyright notice and this permission notice shall be
+  included in all copies or substantial portions of the Software.
+
+  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+  EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+  MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+  NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+  BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+  ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+  CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+  SOFTWARE.
+*/
+
+
+#include <Arduino.h>
+#include <ESP8266WiFi.h>
+
+#include <WiFiManager.h>
+
+#include <debug_helper.h>
+
+#include "WifiConnection.h"
+
+WiFiManager wifiManager;
+
+bool setupWifiStatic(const char *ssid, const char *passwd,
+                     std::function<void()> const &waitCb) {
+  delay(10);
+  waitCb();
+  // We start by connecting to a WiFi network
+  DebugLn();
+  Debug(F("Connecting to "));
+  DebugLn(ssid);
+
+  WiFi.mode(WIFI_STA);  // Explicitly set station mode
+  WiFi.begin(ssid, passwd);
+
+  uint16_t counter = 0;
+
+  while (WiFi.status() != WL_CONNECTED) {
+    waitCb();
+    delay(500);
+    waitCb();
+    delay(500);
+    waitCb();
+    Debug(F("."));
+    if (counter > 180) {
+      DebugLn();
+      DebugLn(F("Could not connect to wifi in 180 seconds!"));
+      return false;
+    }
+    ++counter;
+  }
+
+  DebugLn();
+  DebugLn(F("WiFi connected"));
+  DebugLn(F("IP address: "));
+  DebugLn(WiFi.localIP());
+  waitCb();
+  return true;
+}
+
+bool connectWifi(const char *ssid, const char *passwd,
+                 const std::function<void()> &waitCb) {
```

Comment:
The LED should flash with 1 Hz during connecting to WiFi (see README). I see that this was broken and never worked this way. But now with this single callback and `waitCb=beatLED.loop()` it will only be heartbeat. This is not the idea/function of a status LED, since connecting to wifi, WiFiManager and connected to MQTT are different states. Since WiFiManager did not support a LED callback that is periodically called it is not possible with the `Heartbeat` module. This the reason of HeartbeatFlashing. One idea is to call `beatLED.flash(500)` for the connecting state and `beatLED.flash(250)` for the WiFiManager.